presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -73,12 +74,14 @@ public Analysis analyze(Statement statement, boolean isDescribe)
         analyzer.analyze(rewrittenStatement, Optional.empty());
 
         // check column access permissions for each table
-        analysis.getTableColumnReferences(session.getIdentity())
-                .forEach((tableName, columns) ->
-                        accessControl.checkCanSelectFromColumns(session.getRequiredTransactionId(),
-                                session.getIdentity(),
-                                tableName,
-                                ImmutableSet.copyOf(columns)));
+        analysis.getTableColumnReferences().forEach((identity, tableColumnReferences) -> {
+            AccessControl identityAccessControl = identity.equals(session.getIdentity()) ? accessControl : new ViewAccessControl(accessControl);
```

Comment:
I'm not sure if this is a problem or just needs a comment....  The identity context could be `A->B->A`, and in that case the inner most `A` would be checked with a regular access control instead of a view access control.

Reply:
This logic seems to be intended to be the same as in `StatementAnalyzer.Visitor#analyzeView`, where we do normal access checks for the view if the session user owns it, and I believe the logic should be the same. However, I think it's actually different because `StatementAnalyzer` is called recursively, and thus the session user used when analyzing the innermost `A` would be the owner of `B`. Here, the session user is constant across references.

Reply:
I'm combining the delete and view commits, so basically we explicitly state what access type we're dealing with and use the appropriate access control.  That should fix this issue.